test/javascript/explore-data/search-matrix-spec.js

```diff
@@ -131,17 +146,57 @@ describe("Searching a matrix (table) of data...", function() {
 		});
 
 		it("should survive incomplete input", function() {
-			expect(visualizeMatrix([["t","o"],["b"],["l","o","n","g"]])).toEqual("to\nb\nlong");
+			expect(visualizeMatrix([
+				["t", "o"],
+				["b"],
+				["l", "o", "n", "g"]
+			])).toEqual("to\nb\nlong");
 		});
 
 	});
 
-	it("should run in finite time", function() {
-		expect(true).toBeTruthy();
-	});
+	describe("performance", function() {
+
+		xit("should run in finite time", function() {
+			try {
+				fillMatrix(Infinity);
+				expect(false).toEqual("run out of time");
+			} catch (e) {
+				expect(e.message).toBeDefined();
+				expect(e.message).toContain("Operation is running too slow:");
+			}
+		});
+
+		it("should run in better than N*M time", function() {
+			var startTime;
+			var samples = [];
+			var multiple = 1.5;
+			var size = 500;
+			var k;
+
+			var matrix = []; // without assignment to go faster
+			for (k = 1; k < 5; k++) {
+				startTime = Date.now();
```

Comment:
this takes 4 samples (1,2,3,4) of the fillMatrix function to see if the behavior is exponential ie n^2: 1^2, 2^2, 3^3, 4^4